### PR TITLE
Update pipeline to run tests on Linux

### LINKED ISSUE #441: Java language worker end to end test should also be run on Linux
Java language worker end to end test should also be run on Linux. Currently, the end to end tests are only running on Windows.

### PR DESCRIPTION
**Pull request by shreyas-gopalakrishna**
Update pipeline to run end to end tests on Linux
resolves #441 

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -63,7 +66,79 @@ jobs:
...
+        displayName: 'Install NuGet Tool'
+      - pwsh: |
+          Get-Command mvn
+        displayName: 'Check Maven is installed'
+      - task: JavaToolInstaller@0
+        inputs:
+          versionSpec: '8'
+          jdkArchitectureOption: 'x64'
+          jdkSourceOption: 'PreInstalled'
+        condition: eq( variables['Agent.OS'], 'Linux' )
```
Comment (kaibocai) [inline azure-pipelines.yml:135]: display name? I believe this step is installing jdk8 on the linux vm right.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -73,25 +148,61 @@ jobs:
           curl -sSL https://raw.githubusercontent.com/dotnet/cli/master/scripts/obtain/dotnet-install.sh | bash /dev/stdin -v '6.0.100-rc.2.21505.57' -c 'release' --install-dir /usr/share/dotnet
         displayName: 'Install the .Net version used by the Core Tools for Linux'
         condition: eq( variables['Agent.OS'], 'Linux' )
+      - task: JavaToolInstaller@0
```
Comment (kaibocai) [inline azure-pipelines.yml:152]: is this task repeated with task at line 166?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -73,25 +148,61 @@ jobs:
...
+        inputs:
+          versionSpec: $(JAVA_VERSION_SPEC)
+          jdkArchitectureOption: 'x64'
+          jdkSourceOption: 'PreInstalled'
+        condition: eq( variables['Agent.OS'], 'Linux' )
```
Comment (kaibocai) [inline azure-pipelines.yml:157]: displayName

File: azure-pipelines.yml [hunk 4]
```diff
@@ -73,25 +148,61 @@ jobs:
...
           .\setup-tests-pipeline.ps1
         displayName: 'Setup test environment -- Install the Core Tools'
       - pwsh: |
           $currDir =  Get-Location
-          $Env:Path = $Env:Path+";$currDir\Azure.Functions.Cli"
-          ls $currDir\Azure.Functions.Cli
+          $Env:Path = $Env:Path+";$currDir/Azure.Functions.Cli"
+          ls $currDir/Azure.Functions.Cli
+          func --version
+        condition: eq( variables['Agent.OS'], 'Windows_NT' )
```
Comment (kaibocai) [inline azure-pipelines.yml:188]: displayName

File: azure-pipelines.yml [hunk 5]
```diff
@@ -73,25 +148,61 @@ jobs:
...
+      - bash: |
+          chmod +x ./Azure.Functions.Cli/func
+          chmod +x ./Azure.Functions.Cli/gozip
+          ls ./Azure.Functions.Cli
+          export PATH=$PATH:./Azure.Functions.Cli
           func --version
+        condition: eq( variables['Agent.OS'], 'Linux' )
```
Comment (kaibocai) [inline azure-pipelines.yml:196]: displayName

File: azure-pipelines.yml [hunk 6]
```diff
@@ -117,6 +228,7 @@ jobs:
           projects: |
             endtoendtests\Azure.Functions.Java.Tests.E2E\Azure.Functions.Java.Tests.E2E\Azure.Functions.Java.Tests.E2E.csproj
         env:
+          JAVA_HOME: $(JavaHome)
```
Comment (kaibocai) [inline azure-pipelines.yml]: In this case, are we running this tests only for java8?

File: azure-pipelines.yml [hunk 7]
```diff
@@ -135,96 +247,40 @@ jobs:
...
-        continueOnError: false
-
+          buildType: 'current'
+          downloadType: 'single'
+          artifactName: 'buildDrop'
+          downloadPath: '$(Build.ArtifactStagingDirectory)'
+        displayName: 'Download build artifacts'
+      - pwsh: |
+          Move-Item -Path "$(Build.ArtifactStagingDirectory)/buildDrop/*.nupkg" -Destination $(Build.ArtifactStagingDirectory)
+          Remove-Item "$(Build.ArtifactStagingDirectory)/buildDrop" -Recurse
```
Comment (kaibocai) [inline azure-pipelines.yml:248]: displayName

File: azure-pipelines.yml [hunk 8]
```diff
@@ -73,25 +148,61 @@ jobs:
...
+          export PATH=$PATH:./Azure.Functions.Cli
           func --version
+        condition: eq( variables['Agent.OS'], 'Linux' )
+      - pwsh: |
           cd ./endtoendtests
           mvn clean package `-Dmaven`.javadoc`.skip=true `-Dmaven`.test`.skip `-Dorg`.slf4j`.simpleLogger`.log`.org`.apache`.maven`.cli`.transfer`.Slf4jMavenTransferListener=warn `-B
-          Copy-Item "confluent_cloud_cacert.pem" ".\target\azure-functions\azure-functions-java-endtoendtests"
+          Copy-Item "confluent_cloud_cacert.pem" "./target/azure-functions/azure-functions-java-endtoendtests"
         displayName: 'Package Java for E2E'
       - task: DotNetCoreCLI@2
```
Comment (kaibocai) [inline azure-pipelines.yml:203]: can we run java -version before the test, so we can confirm what exact the java version we are testing against.

File: azure-pipelines.yml (hunk 1 above)
Comment (shreyas-gopalakrishna) [reply azure-pipelines.yml:135]: Yes, to build the java worker

File: azure-pipelines.yml (hunk 2 above)
Comment (shreyas-gopalakrishna) [reply azure-pipelines.yml:152]: No, this is needed to setup Java in Linux for testing. This step installs specific Java version.

File: azure-pipelines.yml (hunk 3 above)
Comment (shreyas-gopalakrishna) [reply azure-pipelines.yml:157]: Skipped displayName as it is under the same previous displayName step.

File: azure-pipelines.yml (hunk 4 above)
Comment (shreyas-gopalakrishna) [reply azure-pipelines.yml:188]: Skipped displayName as it is under the same previous displayName step.

File: azure-pipelines.yml (hunk 5 above)
Comment (shreyas-gopalakrishna) [reply azure-pipelines.yml:196]: Skipped displayName as it is under the same previous displayName step.

File: azure-pipelines.yml (hunk 8 above)
Comment (shreyas-gopalakrishna) [reply azure-pipelines.yml:203]: No, this well not help. Correct measure is to verify the java version from the tests - HttpTriggerJavaVersion

File: azure-pipelines.yml (hunk 2 above)
Comment (kaibocai) [reply azure-pipelines.yml:152]: but we are install the jdk again at line 167 right on Linux

File: azure-pipelines.yml (hunk 6 above)
Comment (shreyas-gopalakrishna) [reply azure-pipelines.yml]: FUNCTIONS_WORKER_JAVA_LOAD_APP_LIBS is only supported in V3. So this test is not useful. Will remove it

File: azure-pipelines.yml (hunk 7 above)
Comment (shreyas-gopalakrishna) [reply azure-pipelines.yml:248]: Skipped displayName as it is under the same previous displayName step.

File: azure-pipelines.yml (hunk 2 above)
Comment (shreyas-gopalakrishna) [reply azure-pipelines.yml:152]: The microsoft JDK download and install is working for windows. For linux, microsoft Jdk download and setting the java home and path is not working. If this step is not added, it uses default java version on linux vm - java 11. [Software information](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#software)
By this step required Java version is installed but it is not microsoft jdk, it is I think open jdk. 

In future, the base image might come with microsoft jdk. This step will automatically use new jdk at that point. 

Coming to downloading and install around line 167, that is again retained until microsoft jdk is available as part of the VM itself OR until we can fix updating the java path which is not currently happening in Linux.

Comment (kaibocai) [reply azure-pipelines.yml:152]: so you mean the task at line 157 is somehow to maker sure task at line 167 to complete successfully?

Comment (shreyas-gopalakrishna) [reply azure-pipelines.yml:152]: Yes, on similar lines. Steps on line 167 are working when tested directly on a Linux machine but not in the pipeline agent/VM.

**Review by kaibocai — APPROVED** [review]
LGTM!

**Pull request merged**